src/Sulu/Bundle/AudienceTargetingBundle/EventListener/DeviceDetectorSubscriber.php

```diff
@@ -39,7 +39,7 @@ public static function getSubscribedEvents()
 
     public function setUserAgent(RequestEvent $event)
     {
-        $this->deviceDetector->setUserAgent($event->getRequest()->headers->get('User-Agent'));
+        $this->deviceDetector->setUserAgent($event->getRequest()->headers->get('User-Agent') ?? '');
```

Comment:
The default can be handled also over the second parameter:

```suggestion
        $this->deviceDetector->setUserAgent($event->getRequest()->headers->get('User-Agent', '');
```